# Exercise 01 - Syntax, Variables and Numbers

Welcome to your first set of Python coding problems! 

**Notebooks** are composed of blocks (called "cells") of text and code. Each of these is editable, though you'll mainly be editing the code cells to answer some questions.

To get started, try running the code cell below (by pressing the `►| Run` button, or clicking on the cell and pressing `ctrl+Enter`/`shift+Enter` on your keyboard).

In [1]:
print("You've successfully run some Python code")
print("Congratulations!")

You've successfully run some Python code
Congratulations!


Try adding another line of code in the cell above and re-running it. 

Now let's get a little fancier:  Add a new code cell by clicking on an existing code cell, hitting the `escape` key *(turn to command mode)*, and then hitting the `a` or `b` key.  
- The `a` key will add a cell above the current cell.
- The `b` adds a cell below.

Great! Now you know how to use Notebooks.

## 0. Creating a Variable

**What is your favorite color? **

To complete this question, create a variable called `color` in the cell below with an appropriate `string` value.

In [2]:
# Create a variable called color with an appropriate value on the line below
# (Remember, strings in Python must be enclosed in 'single' or "double" quotes)

color = "Blue"
print(color)


Blue


<hr/>

## 1. Simple Arithmetic Operation

Complete the code below. In case it's helpful, here is the table of available arithmatic operations:



| Operator     | Name           | Description                                            |
|--------------|----------------|--------------------------------------------------------|
| ``a + b``    | Addition       | Sum of ``a`` and ``b``                                 |
| ``a - b``    | Subtraction    | Difference of ``a`` and ``b``                          |
| ``a * b``    | Multiplication | Product of ``a`` and ``b``                             |
| ``a / b``    | True division  | Quotient of ``a`` and ``b``                            |
| ``a // b``   | Floor division | Quotient of ``a`` and ``b``, removing fractional parts |
| ``a % b``    | Modulus        | Integer remainder after division of ``a`` by ``b``     |
| ``a ** b``   | Exponentiation | ``a`` raised to the power of ``b``                     |
| ``-a``       | Negation       | The negative of ``a``                                  |

<span style="display:none"></span>


In [4]:
pi = 3.14159 # approximate
diameter = 3

# Create a variable called 'radius' equal to half the diameter
radius = diameter/2

# Create a variable called 'area', using the formula for the area of a circle: pi times the radius squared
area = pi*(radius**2)
print(radius)
print(area)

1.5
7.0685775


**Results**:
- Area = 7.0685775

## 2. Variable Reassignment

Add code to the following cell to swap variables `a` and `b` (so that `a` refers to the object previously referred to by `b` and vice versa).

In [11]:
# If you're curious, these are examples of lists. We'll talk about 
# them in depth a few lessons from now. For now, just know that they're
# yet another type of Python object, like int or float.
a = [1, 2, 3]
b = [3, 2, 1]

######################################################################

# Your code goes here. Swap the values to which a and b refer.
# Hint: Try using a third variable

d = b
c = a
b = c
a = d
print(b)
print(a)

[1, 2, 3]
[3, 2, 1]


## 3. Order of Operations


a) Add parentheses to the following expression so that it evaluates to 1.

*Hint*: Following its default "**PEMDAS**"-like rules for order of operations, Python will first divide 3 by 2, then subtract the result from 5. You need to add parentheses to force it to perform the subtraction first.

In [5]:
(5 - 3 )// 2

1

<small>Questions, like this one, marked a spicy pepper are a bit harder. Don't feel bad if you can't get these.</small>

b) <span title="A bit spicy" style="color: darkgreen ">🌶️</span> Add parentheses to the following expression so that it evaluates to **0**.

In [6]:
8 - 3 * 2 - (1 + 1)

0

## 4. Your Turn
Alice, Bob and Carol have agreed to pool their Halloween candies and split it evenly among themselves.
For the sake of their friendship, any candies left over will be smashed. For example, if they collectively
bring home 91 candies, they'll take 30 each and smash 1.

Write an arithmetic expression below to calculate how many candies they must smash for a given haul.

> *Hint*: You'll probably want to use the modulo operator, `%`, to obtain the remainder of division.

In [8]:
# Variables representing the number of candies collected by Alice, Bob, and Carol
alice_candies = 121
bob_candies = 77
carol_candies = 109

# Your code goes here! Replace the right-hand side of this assignment with an expression

# involving alice_candies, bob_candies, and carol_candies
to_smash = (alice_candies+bob_candies+carol_candies)%3
print(to_smash)

1


# Keep Going 💪